File: lab-1/c++/StringManipulator.cpp
/**
* Done by:
 * Student Name: Stanislav Buket
 * Variant: 3
 * Student Group: 121
 * Lab 1.1
 */

#include "StringManipulator.h"
#include <stdexcept>
#include <cctype>

// Приватний метод перевірки, чи містить рядок лише цифри
bool StringManipulator::isDigitsOnly(const std::string& str) const
{
    for (char ch : str)
    {
        if (!std::isdigit(static_cast<unsigned char>(ch)))
        {
            return false;
        }
    }
    return true;
}

// Конструктор з параметром із перевіркою
StringManipulator::StringManipulator(const std::string& value) : value(value)
{
    if (!isDigitsOnly(value))
    {
        throw std::invalid_argument("Рядок має містити лише цифрові символи.");
    }
}

// Гетер для отримання значення рядка
std::string StringManipulator::getValue() const
{
    return value;
}

int StringManipulator::getLength() const
{
    return static_cast<int>(value.length());
}

File: lab-1/c#/StringManipulator.cs
using System;
using System.Linq;

namespace StringManipulatorApp
{
    public class StringManipulator
    {
        private string value;

        // Конструктор з параметром із перевіркою, що рядок містить лише цифри
        public StringManipulator(string value)
        {
            if (!IsDigitsOnly(value))
            {
                throw new ArgumentException("Рядок має містити лише цифрові символи.");
            }
            this.value = value;
        }

        // Властивість для доступу до рядка
        public string Value
        {
            get { return value; }
        }

        public int GetLength()
        {
            return value.Length;
        }

        // Приватний метод для перевірки, що рядок містить лише цифри
        private bool IsDigitsOnly(string str)
        {
            return str.All(char.IsDigit);
        }
    }
}